irohad/consensus/yac/impl/yac_gate_impl.cpp

```diff
@@ -35,104 +35,91 @@ namespace iroha {
             block_loader_(std::move(block_loader)),
             consensus_result_cache_(std::move(consensus_result_cache)),
             log_(logger::log("YacGate")) {
-        block_creator_->on_block().subscribe(
-            [this](auto block) { this->vote(block); });
+        block_creator_->on_block().subscribe([this](auto block) {
+          // TODO(@l4l) 24/09/18 IR-1717
+          // update BlockCreator iface according to YacGate
+          this->vote(nullptr, block, {0, 0});
+        });
       }
 
-      void YacGateImpl::vote(std::shared_ptr<shared_model::interface::Block> block) {
-        auto hash = hash_provider_->makeHash(*block);
-        log_->info("vote for block ({}, {})",
-                   hash.vote_hashes.proposal_hash,
-                   block->hash().toString());
-        auto order = orderer_->getOrdering(hash);
+      void YacGateImpl::vote(
+          std::shared_ptr<shared_model::interface::Proposal> proposal,
```

Comment:
Optionals should be used for both proposal and block, because there could be no proposal from the current ordering service, and therefore no block.

Reply:
Optional sounds redundant, maybe just use (& notice in docs) `nullptr` as no proposa/block case?

Reply:
No, since we guarantee that pointers are not nullable in server code.